src/game/Spell.cpp

```diff
@@ -3185,6 +3185,11 @@ void Spell::SendCastResult(Player* caster, SpellEntry const* spellInfo, SpellCas
 {
     WorldPacket data(SMSG_CAST_FAILED, (4 + 1 + 1));
     data << uint32(spellInfo->Id);
+    
+    if (spellInfo->Id == 24406 && result == SPELL_FAILED_NOTHING_TO_DISPEL) // Improved mend pet - don't send anything to player if nothing to dispel
```

Comment:
This check, being done here, that's not gonna fly. Move it back to checkcast somewhere.